sklearn/utils/validation.py

```diff
@@ -1085,17 +1093,28 @@ def is_sparse(dtype):
                 % (n_features, array.shape, ensure_min_features, context)
             )
 
-    # With an input pandas dataframe or series, we know we can always make the
-    # resulting array writeable:
-    # - if copy=True, we have already made a copy so it is fine to make the
-    #   array writeable
-    # - if copy=False, the caller is telling us explicitly that we can do
-    #   in-place modifications
-    # See https://pandas.pydata.org/docs/dev/user_guide/copy_on_write.html#read-only-numpy-arrays
-    # for more details about pandas copy-on-write mechanism, that is enabled by
-    # default in pandas 3.0.0.dev.
-    if _is_pandas_df_or_series(array_orig) and hasattr(array, "flags"):
-        array.flags.writeable = True
+    if force_writeable:
+        array_data = array.data if sp.issparse(array) else array
+        copy_params = {"order": "K"} if not sp.issparse(array) else {}
+        if hasattr(array_data, "flags") and not array_data.flags.writeable:
```

Comment:
```suggestion
        flags = getattr(array_data, "flags", None)
        if not getattr(flags, "writeable", True):
```

Reply:
Do you think we need to trigger the array API tests again after this fix to be safe ?

Reply:
Let me do that here:

- https://github.com/scikit-learn/scikit-learn/actions/runs/9601131915

Reply:
The failures seem unrelated to this PR. We need to check if they also occur on `main`:

- https://github.com/scikit-learn/scikit-learn/actions/runs/9604238613

Reply:
Confirmed, those failures are not related to this specific PR.